CefSharp.Example/Handlers/AudioHandler.cs

```diff
@@ -33,19 +39,51 @@ protected override void OnAudioStreamStarted(IWebBrowser chromiumWebBrowser, IBr
 
         protected override void OnAudioStreamPacket(IWebBrowser chromiumWebBrowser, IBrowser browser, IntPtr data, int noOfFrames, long pts)
         {
-            //NOTE: data is an array representing the raw PCM data as a floating point type, i.e. 4-byte value(s)
-            //Based on and the channelLayout value passed to IAudioHandler.OnAudioStreamStarted
-            //you can calculate the size of the data array in bytes.
-            //See https://github.com/cefsharp/CefSharp/issues/2806 for discussion on implementing an example.
+            /*
+             * NOTE: data is an array representing the raw PCM data as a floating point type, i.e. 4-byte value(s)
+             * Based on noOfFrames and the channels value passed to IAudioHandler.OnAudioStreamStarted
+             * you can calculate the size of the data array in bytes.
+             * 
+             * Audio data (PCM, 32-bit, float) will be save to rawAudioFile stream.
+             */
+
+            unsafe
+            {
+                float** channelData = (float**)data.ToPointer();
+                int size = channelCount * noOfFrames * 4;
+                byte[] samples = new byte[size];
+                fixed (byte* pDestByte = samples)
+                {
+                    float* pDest = (float*)pDestByte;
+
+                    for (int i = 0; i < noOfFrames; i++)
+                    {
+                        for (int c = 0; c < channelCount; c++)
+                        {
+                            *pDest++ = channelData[c][i];
+                        }
+                    }
+
+                    var ustream = new UnmanagedMemoryStream(pDestByte, size);
```

Comment:
Isn't the data stored in `byte[] samples`? Can we write `samples` to the stream instead of using an `UnmanagedMemoryStream`?

Reply:
Fixed